drivers/sensors/sensor.c

```diff
@@ -609,35 +733,45 @@ static int sensor_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
   FAR struct sensor_upperhalf_s *upper = inode->i_private;
   FAR struct sensor_lowerhalf_s *lower = upper->lower;
   FAR struct sensor_user_s *user = filep->f_priv;
-  int ret;
+  int ret = 0;
 
   sninfo("cmd=%x arg=%08lx\n", cmd, arg);
 
-  ret = nxsem_wait(&upper->exclsem);
-  if (ret < 0)
-    {
-      return ret;
-    }
-
   switch (cmd)
     {
       case SNIOC_GET_STATE:
         {
+          nxrmutex_lock(&upper->lock);
           memcpy((FAR void *)(uintptr_t)arg,
                  &upper->state, sizeof(upper->state));
           user->changed = false;
+          nxrmutex_unlock(&upper->lock);
+        }
+        break;
+
+      case SNIOC_GET_USTATE:
+        {
+          nxrmutex_lock(&upper->lock);
```

Comment:
Why `user->state` access is guarded by `upper->lock`?

Reply:
Since other place(e.g. SNIOC_SET_INTERVAL) which modify field in user->state is locked by upper->lock.

Reply:
Let me re-examine. I think other places involve `upper` together with `user`, so I was expecting that `upper->lock` is for `upper` protection. But maybe I missed something

Reply:
Sensor is a little bit complex, we can't protect through user's lock(actually, user don't have the lock at all), since it may change the hardware.

Reply:
Ok

Reply:
Ok. Lets keep it as is.